scripts/apps/dashboard/widget-react.tsx

```diff
@@ -25,11 +49,20 @@ export class WidgetReact extends React.PureComponent<IProps, IState> {
 
         return (
             <Component
+                ref={this.widgetRef}
                 key={key}
                 article={this.props.article}
                 initialState={(() => {
                     const localStorageWidgetState = JSON.parse(localStorage.getItem('SIDE_WIDGET') ?? 'null');
 
+                    if (localStorageWidgetState == null && closedThroughAction.closed === false) {
```

Comment:
I think you should adjust the order a bit so it's easier to understand. First you should check `closedThroughAction`, and if state needs to be restored you do that first thing and return. Reading local storage should only happen in the "else branch" (or a `return`)

Reply:
This is depending on what we want to prioritise. In theory there can't be a state where we have both `SIDE_WIDGET` and `widgetState` for the same widget, but in practice invalid states can occur. That's why I'm checking if the local storage `SIDE_WIDGET` is empty while also checking `closedThroughAction`. That way I prioritise `SIDE_WIDGET` value over everything.

Reply:
Makes sense, but it would be better to improve the code, so logical order follows code order. If local storage is prioritized, it should be in the code first as well. i.e.

```

if (localStorage) {
    // ...
} else if (restoreState) {
    // ...
} else {
    return undefined;
}
```